src/container/RelayContainer.js

```diff
@@ -128,7 +120,15 @@ function createContainerComponent(
     constructor(props, context) {
       super(props, context);
 
-      var {route} = context;
+      var {relay, route} = context;
+      invariant(
+        relay instanceof RelayContext,
+        'RelayContainer: `%s` was rendered without a valid `relayContext`. ' +
+        'Make sure the `relayContext` is valid, and make sure that it is ' +
+        'correctly set on the parent component\'s context ' +
+        '(e.g. using <RelayRootContainer>).',
+        containerName
+      );
```

Comment:
Nice. The key `relayContext` in the message doesn't match the actual key name, though. Maybe something like 

    'RelayContainer: `%s` was rendered without setting the `relay` property of the React context....'